src/parallel/parallel.hpp

```diff
@@ -53,37 +54,40 @@ class SegmentLock {
     // thread on the lock. However, if the number of locks exceeds an
     // unspecified limit, then it will throw a std::system_error.
     std::recursive_mutex mutex_;
-    // N.B. Using counter_ = 0 causes the linter to complain that the copying
+    // N.B. Using version_ = 0 causes the linter to complain that the copying
     //      invokes a deleted constructor
-    // N.B. Using counter_(0) causes the linter to complain, thinking counter_
-    //      is a function.
-    AtomicCounter counter_{0};
+    // N.B. Using version_(0) will cause the compile to parse version_ as a
+    //      function.
+    // N.B. This needs to be atomic since it can be read from multiple threads
+    //      via get_version() while its being modified by one thread.
+    AtomicVersion version_{0};
+    unsigned locked_count_{0};
```

Comment:
Leave a comment about why `unsigned` is ok. It's because `locked_count_` is bounded

Reply:
Fixed in next commit.